Classes/NSManagedObject+DataMapping.m

```diff
@@ -373,9 +373,9 @@ + (NSDictionary*)processCollectionOfEntitiesAccordingToStatusIndicator:(NSArray*
 + (void) parseAttributes:(NSString*)attributeName mappedItemAttributes:(NSArray*)mappedItemAttributes objAttributes:(NSDictionary*)objAttributes
                      obj:(NSManagedObject*)obj convertedAttributes:(NSMutableDictionary*)convertedAttributes
 {
+    // Only process the key if it's in the remote dictionary, or if it's null (to persist null to the DB).
     NSString *remoteKey = [mappedItemAttributes objectAtIndex:kKernArrayIndexRemoteKey];
-    // only process key if it's in our provided set
-    if ([[objAttributes allKeys] containsObject:remoteKey]) {
+    if ([objAttributes isEqual:[NSNull null]] || ([objAttributes isKindOfClass:[NSDictionary class]] && [[objAttributes allKeys] containsObject:remoteKey])) {
```

Comment:
I think we should skip the function if [objAttributes isKindOfClass:[NSNull class]] . That would act the same as server not returning the prop.